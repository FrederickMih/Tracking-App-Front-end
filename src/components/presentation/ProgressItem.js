import PropTypes from 'prop-types';
import '../../styles/ProgressItem.css';

const formatDate = (string) => {
  const options = { year: 'numeric', month: 'long', day: 'numeric' };
  return new Date(string).toLocaleDateString([], options);
};

const ProgressItem = ({ date, data }) => (
  <div className="progress__item">
    <p className="date">{formatDate(date)}</p>
    <p className="data">
      {data}
      {' '}
      cm
    </p>
  </div>
);

ProgressItem.propTypes = {
  date: PropTypes.string.isRequired,
  data: PropTypes.number.isRequired,
};

export default ProgressItem;
