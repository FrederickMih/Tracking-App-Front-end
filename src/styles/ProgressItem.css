.progress__item {
  font-size: 1rem;
}
