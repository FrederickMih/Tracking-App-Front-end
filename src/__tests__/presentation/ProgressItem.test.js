import React from 'react';
import ProgressItem from '../../components/presentation/ProgressItem';
import renderer from 'react-test-renderer'

it('It should render the ProgressItem correctly', () => {
  const tree = renderer.create(<ProgressItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
