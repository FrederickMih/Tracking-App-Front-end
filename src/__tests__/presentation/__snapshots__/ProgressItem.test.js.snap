// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="progress__item"
>
  <p
    className="date"
  >
    Invalid Date
  </p>
  <p
    className="data"
  >
     
    cm
  </p>
</div>
`;

exports[`It should 1`] = `
<div
  className="progress__item"
>
  <p
    className="date"
  >
    Invalid Date
  </p>
  <p
    className="data"
  >
     
    cm
  </p>
</div>
`;

exports[`It should render the 1`] = `
<div
  className="progress__item"
>
  <p
    className="date"
  >
    Invalid Date
  </p>
  <p
    className="data"
  >
     
    cm
  </p>
</div>
`;

exports[`It should render the ProgressItem correctly 1`] = `
<div
  className="progress__item"
>
  <p
    className="date"
  >
    Invalid Date
  </p>
  <p
    className="data"
  >
     
    cm
  </p>
</div>
`;

exports[`re 1`] = `
<div
  className="progress__item"
>
  <p
    className="date"
  >
    Invalid Date
  </p>
  <p
    className="data"
  >
     
    cm
  </p>
</div>
`;

exports[`renders correctly when there are no items 1`] = `
<div
  className="progress__item"
>
  <p
    className="date"
  >
    Invalid Date
  </p>
  <p
    className="data"
  >
     
    cm
  </p>
</div>
`;
